draft-ietf-quic-transport.md

```diff
@@ -1069,6 +1069,15 @@ to cease using the connection IDs when requested can result in connection
 failures, as the issuing endpoint might be unable to continue using the
 connection IDs with the active connection.
 
+An endpoint SHOULD limit the state it commits to retiring connection IDs using
+RETIRE_CONNECTION_ID by limiting the number of connection IDs that it tracks for
+retirement to at least twice the active_connection_id_limit. An endpoint MUST
```

Comment:
```suggestion
An endpoint SHOULD limit the number of connection IDs for which it has issued an outstanding RETIRE_CONNECTION_ID to at least twice the active_connection_id limit.
```

Reply:
I tried to rewrite the text to make it clearer, but in doing so, I realize that the original text is unclear.

Ignoring the details it says: "It should limit value X to no less than Y."

But I can achieve this by not limiting it at all, which is not the advice we seem to want to give. Is what you are trying to say:

1. There should be a fixed limit.
1. It should be no lower than Y.

What's the rationale for > 2xactive_connection_id.

Reply:
Yes, I think that this is a clearer formulation.  I've taken this and rewritten to:

> An endpoint SHOULD limit the state it commits to retiring connection IDs.  An endpoint SHOULD allow for sending and tracking a number of RETIRE_CONNECTION_ID frames of at least twice the active_connection_id_limit.

These are both recommendations only.  There are reasons why you might not care for these limits.

The 2x is to allow for a combination of you deciding to retire all active connection IDs and the peer asking you to retire all active connection IDs.  If you allow for that much, you will never encounter a situation where the combination of both peers asking for retirement will cause RETIRE_CONNECTION_ID to be dropped or delayed.

That said, this works if you only track 1 RETIRE_CONNECTION_ID at a time, it just might be slow.

Reply:
@ekr, "outstanding RETIRE_CONNECTION_ID" implies the frame is actually in flight, which may not be the case for state management reasons. I think @martinthomson 's edit resolves this problem, so I'm going to commit that one. Hopefully it addresses your concerns.

Reply:
If @ekr is happy I'll resolve this sub-thread.